lib/uplinkc/uplink_definitions.h

```diff
@@ -0,0 +1,126 @@
+#pragma once
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <stdlib.h>
+
+typedef struct APIKey   { long _handle; } APIKey;
+typedef struct Uplink   { long _handle; } Uplink;
+typedef struct Project  { long _handle; } Project;
+typedef struct Bucket   { long _handle; } Bucket;
+typedef struct Map      { long _handle; } Map;
+typedef struct Buffer   { long _handle; } Buffer;
+typedef struct Object   { long _handle; } Object;
+typedef struct Download { long _handle; } Download;
+typedef struct Upload   { long _handle; } Upload;
+
+typedef struct UplinkConfig {
+    struct {
+        struct {
+            uint8_t SkipPeerCAWhitelist;
+        } TLS;
+    } Volatile;
+} UplinkConfig;
+
+typedef struct EncryptionParameters {
```

Comment:
Let's remove the things that have not been implemented yet from here for this PR. That way it's easier to verify later that the alloc/free stuff is implemented for them properly.

Reply:
:+1: